pkg/operator/sync.go

```diff
@@ -244,6 +246,96 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.configClient.ConfigV1().Images().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		klog.Infof("got image.config.openshift.io/cluster, additionalCA: %s", cfg.Spec.AdditionalTrustedCA.Name)
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		}
+		for key := range cm.Data { // default to binay data if it is there, else use data
+			if bd, bdok := cm.BinaryData[key]; bdok {
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			} else if d, dok := cm.Data[key]; dok {
+				raw, err := base64.StdEncoding.DecodeString(d)
```

Comment:
This "try binary data, then base64 or just inject literally" seems weird...do we really need that?  Did you take this logic from somewhere else?

I think the intention is more that things in binarydata need base64 decoding, the other doesn't...

I see similar but different logic in https://github.com/openshift/machine-config-operator/blob/04b528ce55726591134cfffe68dec342742fc113/pkg/operator/sync.go#L1025
which also talks about how we can drop the legacy path there too...

Reply:
I was mimicking what is done with the other certs. I agree on first pass I thought it was strange. I will make this more clear.

Reply:
So, not sure if this is better, but I am thinking of reading both binary and regular data and putting them on disk. I am worried about conflicts in this scenario but I can add a check for matching keys as well.